server2.hs

```diff
@@ -41,19 +41,18 @@ receive h c = forever $ do
 -- <<server
 server :: Handle -> TVar Integer -> TChan String -> IO ()
 server h factor c = do
-  f <- atomically $ readTVar factor     -- <1>
+  f <- readTVarIO factor                -- <1>
```

Comment:
I don't think I mentioned `readTVarIO` in the book, so let's leave this as it is